### PR TITLE
[Numpy]flip

### PR DESCRIPTION
**Pull request by tingying2020**
## Description ##
operator flip

### CODE REVIEW

File: src/operator/numpy/np_flip_op.cc [hunk 1]
```diff
@@ -0,0 +1,67 @@
+/*
```
Comment (haojin2) [inline src/operator/numpy/np_flip_op.cc]: Move the code to the existing `np_matrix_op*` files.

File: tests/python/unittest/test_numpy_op.py [hunk 2]
```diff
@@ -829,6 +829,71 @@ def get_indices(axis_size):
...
+    for hybridize in [True, False]:
+        for oneType in types:
+            if oneType is 'float16':
+                rtol=1e-2
+                atol=1e-2
+            else:
+                rtol=1e-3
+                atol=1e-5
+            for shape in shapes1:
+                for axis in [None]:
```
Comment (haojin2) [inline tests/python/unittest/test_numpy_op.py]: If you just have 1 thing to test then do not have a for-loop, just do 
```Python
for shape in shapes1:
    axis = None
    # your test code
```

File: tests/python/unittest/test_numpy_op.py [hunk 3]
```diff
@@ -829,6 +829,71 @@ def get_indices(axis_size):
                     assert_almost_equal(mx_out.asnumpy(), np_out, rtol=1e-3, atol=1e-5)
 
 
+@with_seed()
+@use_np
+def test_np_flip():
+    class TestFlip(HybridBlock):
+        def __init__(self, axis):
+            super(TestFlip, self).__init__()
+            # Necessary initialization
```
Comment (haojin2) [inline tests/python/unittest/test_numpy_op.py]: Get rid of this line.

File: tests/python/unittest/test_numpy_op.py [hunk 4]
```diff
@@ -829,6 +829,71 @@ def get_indices(axis_size):
...
+            return F.np.flip(x, self.axis)
+
+    shapes1 = [(1, 2, 3), (1, 0), (3, 0, 2), (), (1,)]
+    shapes2 = [(2, 2, 3, 3), (1, 1, 2, 4)]
+    types = ['int32', 'int64', 'float16', 'float32', 'double']
+    for hybridize in [True, False]:
+        for oneType in types:
+            if oneType is 'float16':
+                rtol=1e-2
+                atol=1e-2
```
Comment (haojin2) [inline tests/python/unittest/test_numpy_op.py]: why do you need higher tolerance for `float16` type? `flip` is a kind of copy, there should be no numerical instability issues.

File: tests/python/gpu/test_operator_gpu.py [hunk 5]
```diff
@@ -2331,6 +2331,72 @@ def test_math():
             for op in ops:
                 run_math(op, shape, dtype, check_value=check_value)
 
+
+@with_seed()
+@use_np
+def test_np_flip():
```
Comment (haojin2) [inline tests/python/gpu/test_operator_gpu.py]: You do not need to add the same test once again in `test_operator_gpu.py` there's an `import` up above already. Delete this function.

File: tests/python/unittest/test_numpy_op.py (hunk 2 above)
Comment (tingying2020) [reply tests/python/unittest/test_numpy_op.py]: Fixed, thank you.

File: tests/python/unittest/test_numpy_op.py (hunk 3 above)
Comment (tingying2020) [reply tests/python/unittest/test_numpy_op.py]: Fixed, thank you.

File: tests/python/unittest/test_numpy_op.py (hunk 4 above)
Comment (tingying2020) [reply tests/python/unittest/test_numpy_op.py]: Fixed, thank you.

File: tests/python/gpu/test_operator_gpu.py (hunk 5 above)
Comment (tingying2020) [reply tests/python/gpu/test_operator_gpu.py]: Done, thank you.

File: src/operator/numpy/np_matrix_op-inl.h [hunk 6]
```diff
@@ -60,6 +60,82 @@ void NumpyTranspose(const nnvm::NodeAttrs& attrs,
...
+#define FLIP_MIN_DIM -1
+
+template<typename xpu>
+void NumpyFlipForwardImpl(const OpContext& ctx,
+                          const std::vector<TBlob>& inputs,
+                          const std::vector<TBlob>& outputs,
+                          const std::vector<index_t>& stride_,
+                          const std::vector<index_t>& trailing_,
+                          const index_t& flip_index);
+
```
Comment (haojin2) [inline src/operator/numpy/np_matrix_op-inl.h]: get rid of the extra blank line

Comment (tingying2020) [reply src/operator/numpy/np_matrix_op-inl.h]: Done. Thank you.

File: src/operator/numpy/np_flip_op.cc (hunk 1 above)
Comment (tingying2020) [reply src/operator/numpy/np_flip_op.cc]: Done.

**Pull request merged**